ocaml/xapi/xapi_vgpu_type.ml

```diff
@@ -363,6 +363,30 @@ module Vendor_nvidia = struct
   let pt_when_vgpu = true
   let whitelist_file () = !Xapi_globs.nvidia_whitelist
   let device_id_of_conf conf = conf.identifier.Identifier.pdev_id
+  
+  let host_driver_version = try Scanf.sscanf (Unix.readlink !Xapi_globs.nvidia_host_driver_file) "libnvidia-vgpu.so.%s" (fun x -> x)
+    with _ -> info "can not get the file version of the %s, use the default value" !Xapi_globs.nvidia_host_driver_file; Xapi_globs.nvidia_default_host_driver_version
+
+  let is_host_driver_support_multi_vgpu () =
+    List.mem host_driver_version !Xapi_globs.nvidia_multi_vgpu_enabled_driver_versions ||
+    (
+      let from_version_on_mark = '+' in
+      let host_driver_major_version, host_driver_minor_version = try Scanf.sscanf host_driver_version "%d.%d" (fun x y -> x , y)
+        with _ -> info "get error when parse the host driver version: %s" host_driver_version;
+          Scanf.sscanf default_host_driver_version "%d.%d" (fun x y -> x , y) in (* If cannot parse host driver version, we use the default one,
+                                                                                    which would be taken as not support multiple vGPU *)
+      !Xapi_globs.nvidia_multi_vgpu_enabled_driver_versions
```

Comment:
Would it be easier to solve this using regular expressions? You can easily check the general format and extract the version numbers.

```
let rx' = Re.(compile @@ Pcre.re {|([0-9]+)\.([0-9]+)(\+?)|})
Re.Pcre.extract rx' "5.6+"
- : string array = [|"5.6+"; "5"; "6"; "+"|]
```

Reply:
thanks, my utop does not have `Re` module, I will figure that how that works and change accordly.

Reply:
updated